modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -280,11 +280,11 @@ public function order_process( $order_id ) {
 
 		// loop through products in the order and queue a purchase event.
 		foreach ( $order->get_items() as $order_item_id => $order_item ) {
-			$product = $order->get_product_from_item( $order_item );
+			$product = is_callable( array( $order_item, 'get_product' ) ) ? $order_item->get_product() : null;
```

Comment:
Nit: could we get directly the `id` here so there is no need to call `$product->get_id()` below?
```suggestion
			$product_id = is_callable( array( $order_item, 'get_product_id' ) ) ? $order_item->get_product_id() : -1;
```

Reply:
Good spotting - have made this change and retested. 👍